### PR TITLE
add uri encoding for mongodb connector password🐛

### LINKED ISSUE #10569: Source MongoDB: connector fail with password which contains special character
Dears
I found one bug when creating MongoDB connector with password which contains special character such as %
## Environment
- **Airbyte version**:  0.35.3-alpha
- **OS Version / Instance**: Centos7
- **Deployment**:  Docker
- **Source Connector and version**: source-mongodb-v2  0.1.10
- **Severity**:  Low
- **Step where error happened**: Create connector

## Current Behavior
It appears error when creating MongoDB source connector with password which contains special character such as % + / ?

## Expected Behavior
Password can contain special characters, for example % + / ?

## Steps to Reproduce
1. Create MongoDB connector
2. Input all necessary information  
3. Input password which consist of special character

## Are you willing to submit a PR?
Currently, do not have this plan.

### PR DESCRIPTION
**Pull request by aglucky**
## What
Fixes #10569

MongoDB passwords need to be encoded with URI encoding to be valid. This is shown in the below image.

image

## How

I encoded an encoding step when creating the config instance variable for the connector using the Ruby URI module.

## Testing

There were no existing integration tests and I am unsure how to add them in this situation.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello @aglucky thanks for the contribution. I added to the source team backlog to review it. At the moment there are quite a lot of development been made for the Mongo destination and source connector and maybe your change will be contemplated for this refactor.

File: airbyte-integrations/connectors/source-mongodb-v2/metadata.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ data:
   tags:
     - language:java
   releases:
+    1.0.1:
+      message: >
+        Patched bug where passwords with special characters were not being handled correctly.
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mongodb-v2/metadata.yaml:27]: This should go to changelog in [mongodb-v2.md](https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/mongodb-v2.md?plain=1)

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java [hunk 2]
```diff
@@ -40,7 +42,7 @@ public static MongoClient createMongoClient(final MongoDbSourceConfig config) {
     if (config.hasAuthCredentials()) {
       final String authSource = config.getAuthSource();
       final String user = config.getUsername();
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java:44]: Does user need encoding as well?

Comment (aglucky) [reply airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java:44]: [It appears so](https://www.mongodb.com/docs/manual/reference/connection-string/). I didn't realize usernames were allowed to have special characters in the first place.

Comment (rodireich) [reply airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java:44]: It doesn't look like any of the class that generate the client connection string take care of encoding. 
great stuff!

Comment (aglucky) [reply airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoConnectionUtils.java]: Thanks, just addressed the feedback!

**Review by rodireich — APPROVED** [review]

**Comment by aglucky** [comment]
@rodireich Just pushed the update to dockerImageTag

**Pull request closed without merging**